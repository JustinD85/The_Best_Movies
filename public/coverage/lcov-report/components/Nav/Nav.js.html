<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Nav/Nav.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Nav</a> Nav.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/2</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import { NavLink } from 'react-router-dom'
import './Nav.scss'
&nbsp;
const Nav = () =&gt; {
  return (
    &lt;nav className='Nav'&gt;
      &lt;h1 className='movie-tracker-title'&gt;Movie &lt;span&gt;Tracker&lt;/span&gt;&lt;/h1&gt;
      &lt;div className='nav-border'&gt;
        &lt;div className='nav-options'&gt;
          &lt;p className='browse'&gt;SORT BY:&lt;/p&gt;
          &lt;NavLink exact to='/' activeStyle={{ color: '#0CD0FC' }} name='recent'&gt;RECENT&lt;/NavLink&gt;
          &lt;NavLink exact to='/favorites' activeStyle={{ color: '#0CD0FC' }} name='favorites'&gt;FAVORITES&lt;/NavLink&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/nav&gt;
  )
}
&nbsp;
export default Nav</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 06 2019 09:52:14 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
