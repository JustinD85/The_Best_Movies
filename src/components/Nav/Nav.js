import React, { Fragment } from 'react';
import { Route, NavLink } from 'react-router-dom';
import { setFilter, setMovies } from '../../actions';
import { connect } from 'react-redux';
import apiKey from '../../API_KEY/movie_api_key'



const Nav = (props) => {
  let movies;
  const fetchMovies = async () => {
      const response = await fetch(`https://api.themoviedb.org/3/movie/upcoming${apiKey}`);
      const movies = await response.json();
      console.log('winning',movies)
    await props.setMovies(movies)
    return movies
  }

  return < div onClick={fetchMovies}>{`${JSON.stringify(props.movies)}`} Test</div >
    /* <nav>
      <NavLink to='/recent'  name='recent'>Test {props.movies}</NavLink>
      <NavLink to='/popular' name='popular' onClick={(e) => {
        e.preventDefault()
        props.setFilter(e.target.name)
      }}>Popular Movies</NavLink>
      <NavLink to='/favorites' name='favorites' onClick={(e) => {
        e.preventDefault()
        props.setFilter(e.target.name)
      }}>Favorite Movies</NavLink>
    </nav> */
  
    /* <Route path='/recent' component={Home} /> */

    }

const mapDispatchToProps = (dispatch) => ({
  setFilter: (text) => dispatch(setFilter(text)),
  setMovies: (movies) => dispatch(setMovies(movies)),
})

const mapStateToProps = (state) => ({
  movies: state.movies
})

export default connect(mapStateToProps, mapDispatchToProps)(Nav);