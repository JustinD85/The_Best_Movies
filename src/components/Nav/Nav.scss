nav {
  background: #141d25;
  height: 100vh;
  width: 20%;

  .movie-tracker-title {
    font-family: 'Pacifico', sans-serif;
    color: #ECEBEE;
    letter-spacing: 2px;
    text-align: left;
    padding-left: 35px;

    span {
      padding-left: 20px;
      display: block;
    }
  }

  .nav-border {
    .nav-options {
      display: flex;
      flex-direction: column;
      margin-top: 50%;
  
      .browse {
        color: #ECEBEE;
        text-align: left;
        border-bottom: .5px solid #3e4a56;
        padding-left: 35px;
        letter-spacing: 1px;
        padding-bottom: 11px;
        margin-bottom: 20px;
      }
  
      a {
        text-align: left;
        letter-spacing: 2px;
        border-left: 3px solid transparent;
        font-weight: 500;
        align-items: center;
        color: #ECEBEE;
        margin: 10px auto;
        text-decoration: none;
        width: 100%;
        height: 22px;
        padding-left: 35px;
        font-size: 18px;
      }
    
      a:hover {
        border-left: 3px solid #55E4F5;
      }
    }
  }

}