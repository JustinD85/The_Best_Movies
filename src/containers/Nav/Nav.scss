nav{
  background:#232D32;
  height:100vh;
  display:grid;
  width:10%;
  align-items: center;
  a{
    color:#C4C4C4;
  }
}