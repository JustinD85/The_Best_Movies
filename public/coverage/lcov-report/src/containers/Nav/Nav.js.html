<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/containers/Nav/Nav.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/containers/Nav</a> Nav.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">23.08% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">27.27% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/11</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Fragment } from 'react';
import { NavLink } from 'react-router-dom';
import { setFilter, setMovies } from '../../actions';
import { connect } from 'react-redux';
import apiKey from '../../API_KEY/movie_api_key';
import './Nav.scss'
&nbsp;
const Nav = <span class="fstat-no" title="function not covered" >(p</span>rops) =&gt; {
&nbsp;
  const fetchMovies = <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >as</span>ync () =&gt; {</span>
    const response = <span class="cstat-no" title="statement not covered" >await fetch(`https://api.themoviedb.org/3/movie/upcoming${apiKey}`);</span>
    const movies = <span class="cstat-no" title="statement not covered" >await response.json();</span>
<span class="cstat-no" title="statement not covered" >    await props.setMovies(movies);</span>
<span class="cstat-no" title="statement not covered" >    return movies;</span>
  }
&nbsp;
<span class="cstat-no" title="statement not covered" >  return (</span>
    &lt;nav className='Nav'&gt;
      &lt;button onClick={fetchMovies}&gt; Click Me and check Redux State&lt;/button &gt;
&nbsp;
      &lt;NavLink to='/recent' name='recent'&gt;Recent&lt;/NavLink&gt;
      &lt;NavLink to='/popular' name='popular'&gt;Popular Movies&lt;/NavLink&gt;
      &lt;NavLink to='/favorites' name='favorites'&gt;Favorite Movies&lt;/NavLink&gt;
&nbsp;
    &lt;/nav&gt;)
&nbsp;
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  movies: state.movies
})
&nbsp;
const mapDispatchToProps = <span class="fstat-no" title="function not covered" >(d</span>ispatch) =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  setFilter: <span class="fstat-no" title="function not covered" >(t</span>ext) =&gt; <span class="cstat-no" title="statement not covered" >dispatch(setFilter(text)),</span>
  setMovies: <span class="fstat-no" title="function not covered" >(m</span>ovies) =&gt; <span class="cstat-no" title="statement not covered" >dispatch(setMovies(movies)),</span>
})
&nbsp;
export default connect(mapStateToProps, mapDispatchToProps)(Nav);</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jan 30 2019 20:31:54 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
