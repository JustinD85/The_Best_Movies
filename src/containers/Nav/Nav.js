import React from 'react'
import { NavLink } from 'react-router-dom'
// import { setFilter } from '../../actions'
// import { connect } from 'react-redux'
import './Nav.scss'

const Nav = () => {
  return (
    <nav className='Nav'>
      <NavLink to='/recent' name='recent'>Recent</NavLink>
      <NavLink to='/popular' name='popular'>Popular Movies</NavLink>
      <NavLink to='/favorites' name='favorites'>Favorite Movies</NavLink>
    </nav>
  )
}

// const mapStateToProps = (state) => ({
//   movies: state.movies
// })

// const mapDispatchToProps = (dispatch) => ({
  // setFilter: (text) => dispatch(setFilter(text)),
  // setMovies: (movies) => dispatch(setMovies(movies)),
// })

// export default connect(mapStateToProps, mapDispatchToProps)(Nav)

export default Nav