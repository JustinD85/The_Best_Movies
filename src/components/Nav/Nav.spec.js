import React from 'react'
import { shallow } from 'enzyme'
import Nav from './Nav'

let wrapper = shallow(<Nav/>)
it('it should match the snapshot', () => {
  expect(wrapper).toMatchSnapshot()
})