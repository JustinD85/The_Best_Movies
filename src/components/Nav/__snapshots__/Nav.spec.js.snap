// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match the snapshot 1`] = `
<nav
  className="Nav"
>
  <h1
    className="movie-tracker-title"
  >
    Movie 
    <span>
      Tracker
    </span>
  </h1>
  <div
    className="nav-border"
  >
    <div
      className="nav-options"
    >
      <p
        className="browse"
      >
        SORT BY:
      </p>
      <NavLink
        activeClassName="active"
        activeStyle={
          Object {
            "color": "#0CD0FC",
          }
        }
        aria-current="page"
        exact={true}
        name="recent"
        to="/"
      >
        RECENT
      </NavLink>
      <NavLink
        activeClassName="active"
        activeStyle={
          Object {
            "color": "#0CD0FC",
          }
        }
        aria-current="page"
        exact={true}
        name="favorites"
        to="/favorites"
      >
        FAVORITES
      </NavLink>
    </div>
  </div>
</nav>
`;
